trino/auth.py

```diff
@@ -546,17 +546,15 @@ def _construct_cache_key(host: Optional[str], user: Optional[str]) -> Optional[s
 
     @staticmethod
     def _parse_authenticate_header(header: str) -> Dict[str, str]:
-        split_challenge = header.split(" ", 1)
-        trimmed_challenge = split_challenge[1] if len(split_challenge) > 1 else ""
+        components = header.split(", ")
```

Comment:
is the whitespace guaranteed to be present?

Reply:
In the context of the WWW-Authenticate header as defined by RFC 7235 (which governs HTTP Authentication) and RFC 2616 for HTTP/1.1, the comma followed by a space is not strictly required. Both a simple comma (,) and a comma followed by a space (, ) are acceptable, though the latter is commonly used for readability.